repository/AdminRepository.js

```diff
@@ -47,8 +61,73 @@ class AdminRepository {
 
     async approveDoctor(doctorId, adminId) {
         try {
-            this.verifyAdmin(adminId);
-            return await this.mongo.db('test').collection('user').updateOne({ userId: doctorId }, { $set: { userStatus: 'APPROVED' } });
+            await this.verifyAdmin(adminId);
+            return await this.mongo.db('test').collection('user').updateOne({ uid: doctorId }, { $set: { userStatus: 'APPROVED' } });
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    async rejectDoctor(doctorId, adminId) {
+        try {
+            await this.verifyAdmin(adminId);
+            return await this.mongo.db('test').collection('user').updateOne({ uid: doctorId }, { $set: { userStatus: 'REJECTED' } });
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    async fetchPendingHealthOfficers(adminId) {
+        try {
+            await this.verifyAdmin(adminId);
+            const response = await this.mongo.db('test').collection('user').find({ userType: 'healthOfficer', userStatus: 'PENDING' }, { name: 1, email: 1 });
+            return response.toArray();
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    async approveHealthOfficer(officerId, adminId) {
+        try {
+            await this.verifyAdmin(adminId);
+            return await this.mongo.db('test').collection('user').updateOne({ uid: officerId }, { $set: { userStatus: 'APPROVED' } });
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    async rejectHealthOfficer(officerId, adminId) {
+        try {
+            await this.verifyAdmin(adminId);
+            return await this.mongo.db('test').collection('user').updateOne({ uid: officerId }, { $set: { userStatus: 'REJECTED' } });
```

Comment:
To keep things DRY, we could have 1 action for all the user to save their info on the database after REJECTED and ACCEPTED, since it's the same logic for all the user type.

Reply:
The same thing could be said for fetchPending*(). It's the same logic copy pasted. At the minimum, you could extract the logic and reuse it.